amqp.go

```diff
@@ -100,8 +129,11 @@ func NewRabbitChannel(parentCtx context.Context, wg *sync.WaitGroup, cfg *Config
 		logrus.Debug("RabbitMQ.URL: ", url)
 	}
 
-	ch.ctx = context.WithValue(parentCtx, rabbitCtxKey, nil)
-	ch.cancel = make(chan bool)
+	ctx := context.WithValue(parentCtx, rabbitCtxKey, nil)
+	ctx, cancel := context.WithCancel(ctx)
+	ch.ctx = ctx
+	ch.cancelFunc = cancel
```

Comment:
I think this cancel function needs to be passed from outside, otherwise its just a internal context which cant be controlled by the service using amqpwrapper